safe-global/safe-react
# refactor: cleanup notifications system

## What it solves
An introduction to https://github.com/safe-global/safe-pm/issues/83

## How this PR fixes it
The current notification system has been refactored:

- [Notistack](https://github.com/iamhosseindhv/notistack) upgraded to the newest MUI4-supporting version.
- System has been simplified: the new actions are `showNotification`, `close/closeAllNotification(s)` and `delete/deleteAllNotification(s)` and the syntax follows Notistack as closely as possible.
- Immutable has been removed from the `notifications` store.
- Notifications are now cached for later retrieval.

None of the above adjustments should yet reflect/alter the UI in it's current state.

## How to test it
Interact with the Safe and observe notifications functioning as before.

One notable notification to check, which includes an `onClick` handler, is dispatched when the Safe requires an update. I removed some unnecessary code that is responsible for address book interaction notifications. They should still work as before as well.
